src/commands/terraform.ts

```diff
@@ -9,13 +9,12 @@ import { LanguageClient } from 'vscode-languageclient/node';
 import * as terraform from '../api/terraform/terraform';
 
 export class TerraformCommands implements vscode.Disposable {
-  private commands: vscode.Disposable[];
-
   constructor(
     private client: LanguageClient,
     private reporter: TelemetryReporter,
+    private context: vscode.ExtensionContext,
   ) {
-    this.commands = [
```

Comment:
In each of these dispose patterned classes eslint warned of "unsafe return of value of type any', and IIRC a possible unhandled exception in a dispose method.

I think we have to examine the command classes and features separately. The features are controlled by the StaticFeature lifetime, which to my understanding of documentation are kinda like singletons. Where as the commands are controlled by 'us', and are supposed to run the lifetime of the extension.

I may be over thinking, but ExtensionContext.subscriptions will dispose at deactivation and so we want all things (like command handlers) to sit until extension is deactivated. Whereas a StaticFeature is registered at LSP initialization and may call it's methods at undetermined times, and the dispose method is not part of StaticFeature's signature so it won't call it, we are relying on things still being "alive" when the extension deactivates.

This may be far more thought than needed for this, but it looked to me like less code that left the disposal time to the framework and not to us.

Reply:
I agree. From the types, it looks like `StaticFeature` doesn't need to implement `dispose`, so maybe we can remove all `dispose` methods from the static features?

Our commands currently implement `vscode.Disposable`, but if we empty the `dispose` method, we could also remove the `implements vscode.Disposable` and `dispose` methods for those as well.

Reply:
This might be a sign they should be functions rather than classes

Reply:
Going to resolve this in a separate PR